src/DotLiquid.Tests/FilterTests.cs

```diff
@@ -206,6 +267,111 @@ public void TestFilterWithMultipleMethodSignaturesAndContextParam()
+            _context.AddFilters(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesTwo));
+
+            Assert.AreEqual("AB", new Variable("'A' | concatenate : 'B'").Render(_context));
+            Assert.AreEqual("ABC", new Variable("'A' | concatenate : 'B', 'C'").Render(_context));
+        }
+
+        [Test]
+        public void TestFilterAsLocalFilterWithMultipleMethodSignaturesDifferentClasses()
+        {
+
+            Helper.AssertTemplateResult(
+                expected: "AB // ABC",
+                template: "{{'A' | concatenate : 'B'}} // {{'A' | concatenate : 'B', 'C'}}",
+                localVariables: null,
+                localFilters: new[] { typeof(FiltersWithMultipleMethodSignaturesDifferentClassesOne), typeof(FiltersWithMultipleMethodSignaturesDifferentClassesTwo) });
+        }
+
+
+        [Test]
+        public void TestFilterWithMultipleMethodSignaturesAndContextParamInDifferentClasses()
+        {
+            Template.RegisterFilter(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne));
+            Template.RegisterFilter(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo));
+
+            Assert.AreEqual("AB", Template.Parse("{{'A' | concat_with_context : 'B'}}").Render());
+            Assert.AreEqual("ABC", Template.Parse("{{'A' | concat_with_context : 'B', 'C'}}").Render());
+        }
+
+        [Test]
+        public void TestFilterAsLocalFilterWithMultipleMethodSignaturesAndContextDifferentClasses()
+        {
+
+            Helper.AssertTemplateResult(
+                expected: "AB // ABC",
+                template: "{{'A' | concat_with_context : 'B'}} // {{'A' | concat_with_context : 'B', 'C'}}",
+                localVariables: null,
+                localFilters: new[] { typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne), typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo) });
+        }
+
+
+        [Test]
+        public void TestFilterInContextWithMultipleMethodSignaturesAndContextParamInDifferentClasses()
+        {
+            _context.AddFilters(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamOne));
+            _context.AddFilters(typeof(FiltersWithMultipleMethodSignaturesDifferentClassesWithContextParamTwo));
+
+            Assert.AreEqual("AB", new Variable("'A' | concat_with_context : 'B'").Render(_context));
+            Assert.AreEqual("ABC", new Variable("'A' | concat_with_context : 'B', 'C'").Render(_context));
+        }
+
+        [Test]
+        public void TestFilterOverridesMethodWithSameMethodSignaturesDifferentClasses()
+        {
+            Template.RegisterFilter(typeof(FilterWithSameMethodSignatureDifferentClassOne));
+            Template.RegisterFilter(typeof(FilterWithSameMethodSignatureDifferentClassTwo));
+
+            Assert.AreEqual("ABClass Two", Template.Parse("{{'A' | concatenate : 'B'}}").Render());

```

Comment:
Please add a code comment explaining why this order of preference is happening when the method signatures are the same.

Reply:
Fixed: https://github.com/dotliquid/dotliquid/pull/453/commits/80e1366110c8a64f680dd1cccfd2c344dd10effc